jwk/daemon_test.go

```diff
@@ -177,8 +177,8 @@ func Test_jwkd_Start(t *testing.T) {
 						return errors.New("cannot update keys")
 					}
 
-					if k1 == k2 {
```

Comment:
It is more accrue to compare the pointer rather then the key ID (check if the pointer is changed or not)

Reply:
new jwk.Set is created for each fetch request, so we need to check the content

Reply:
Actually it create a new object and store into atomic.Value so I think it is ok. 
This test case is check if the j.key can be refreshed or not by the timer.

Reply:
I just want to be explicit about the test case.
As updating the object is the default behavior, we cannot imply changing of the object as the actual update is done. Explicit checking for the content inside the object is required.

Reply:
It makes sense. LGTM.